tests/basic/storage_pool_test.go

```diff
@@ -7923,3 +7923,223 @@ var _ = Describe("{DiffPoolExpansionFromMaintenanceNode}", func() {
+		StartTorpedoTest("DriveAddAsJournal",
+			"Add drive when as journal",
+			nil, testrailID)
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+	var contexts []*scheduler.Context
+	stepLog := "Add drive when as journal"
+	It(stepLog, func() {
+		log.InfoD(stepLog)
+
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("adddriveasjournal-%d", i))...)
+		}
+		ValidateApplications(contexts)
+		defer appsValidateAndDestroy(contexts)
+
+		// Get Pool with running IO on the cluster
+		poolUUID, err := GetPoolIDWithIOs(contexts)
+		log.FailOnError(err, "Failed to get pool running with IO")
+		log.InfoD("Pool UUID on which IO is running [%s]", poolUUID)
+
+		// Get Node Details of the Pool with IO
+		nodeDetail, err := GetNodeWithGivenPoolID(poolUUID)
+		log.FailOnError(err, "Failed to get Node Details from PoolUUID [%v]", poolUUID)
+		log.InfoD("Pool with UUID [%v] present in Node [%v]", poolUUID, nodeDetail.Name)
+
+		startDriver := func() {
+			// Start Px Back on the Node
+			log.InfoD("Start Px Driver and wait for driver to come up on node [%v]", nodeDetail.Name)
+			log.FailOnError(Inst().V.StartDriver(*nodeDetail),
+				fmt.Sprintf("Failed to Bring back the Px on Node [%v]", nodeDetail.Name))
+			log.FailOnError(Inst().V.WaitDriverUpOnNode(*nodeDetail, addDriveUpTimeOut),
+				fmt.Sprintf("Driver is still down on node [%v] after waiting", nodeDetail.Name))
+		}
+		// Bring Px Down on the Node selected
+		var nodeToPxDown []node.Node
+		nodeToPxDown = append(nodeToPxDown, *nodeDetail)
+		log.FailOnError(Inst().V.StopDriver(nodeToPxDown,
+			false,
+			nil),
+			"Errored while stopping Px Driver")
+
+		// wait for some time for driver to go down completely
+		log.FailOnError(Inst().V.WaitDriverDownOnNode(*nodeDetail), "Failed waiting for driver to come up")
+
+		// Start PxDriver after attempting add cloud drive
+		defer startDriver()
+		path := "/etc/pwx/config.json"
+		cmd := "sudo cat " + path
+		out, err := Inst().N.RunCommandWithNoRetry(*nodeDetail, cmd, node.ConnectionOpts{
+			Timeout:         2 * time.Minute,
+			TimeBeforeRetry: 10 * time.Second,
+		})
+		// Add cloud drive on the node selected and wait for rebalance to happen
+		driveSpecs, err := GetCloudDriveDeviceSpecs()
+		log.FailOnError(err, "Error getting cloud drive specs")
+
+		deviceSpec := driveSpecs[0]
+		devicespecjournal := deviceSpec + " --journal"
+		if strings.Contains(out, `"type": "dmthin",`) {
+			err := Inst().V.AddCloudDrive(nodeDetail, devicespecjournal, -1)
+			re := regexp.MustCompile(".*Journal/Metadata device add not supported for PX-StoreV2*")
+			if re.MatchString(fmt.Sprintf("%v", err)) == false {
+				log.Error("Failed to match the error while adding drive")
+			}
+			log.InfoD(fmt.Sprintf("Errored while adding Pool as expected on Node [%v]", nodeDetail.Name))
+		} else {
+			err = Inst().V.EnterPoolMaintenance(*nodeDetail)
```

Comment:
I dont think pool maintenance is required here

Reply:
Pool maintenance is required

Reply:
can you please check, we are doing add cloud drive in other tests without entering pool maintenance and it is working fine

Reply:
This is journal device add and it requires pool maintenance